Assets/Puzzle3.cs

```diff
@@ -0,0 +1,175 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using System.Linq;
+using UnityEngine.Rendering;
+
+public class Puzzle3 : MonoBehaviour
+{
+    [SerializeField]
+    public Button[] oliverButtons;
+    [SerializeField]
+    public Button[] wendyButtons;
+    [SerializeField]
+    public GameObject BasementDoor;
+
+    public string correctOliver = "OLIVER";
+    public string correctWendy = "WENDY";
+
+    private Button selectedButton = null;
+
+    private void Start()
+    {
+        Debug.Log("Oliver Buttons Count: " + oliverButtons.Length);
+        Debug.Log("Wendy Buttons Count: " + wendyButtons.Length);
+
+        foreach (Button button in oliverButtons)
+        {
+            button.onClick.AddListener(() => OnButtonClicked(button));
+        }
+        foreach (Button button in wendyButtons)
+        {
+            button.onClick.AddListener(() => OnButtonClicked(button));
+        }
+    }
+
+    private void Update()
+    {
+        Debug.Log("Accessing Update");
+        var cameraRay = Camera.main.ScreenPointToRay(new Vector2(Screen.width/2, Screen.height/2));
+        Debug.DrawRay(cameraRay.origin, cameraRay.direction * 10000, Color.magenta);
+
+        RaycastHit hit;
+        if (Physics.Raycast(cameraRay, out hit, 10000f))
+        {
+            Debug.Log("Raycast hit UI");
+            Debug.Log("Game Object hit: " + hit.transform.gameObject.name);
+
+            if (Input.GetMouseButtonDown(0))
```

Comment:
I'm going to suggest we move this to the first part of this update handling. We only want to do the raycast when the use clicks instead of every frame to reduce the costs since it is only used when clicked anyways.